src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs

```diff
@@ -26,11 +23,21 @@ private MessageHandler(Type serviceType, Func<object> createMessageHandlerImplem
         {
             Guard.NotNull(serviceType, nameof(serviceType));
             Guard.NotNull(createMessageHandlerImplementation, nameof(createMessageHandlerImplementation));
+            Guard.For<ArgumentException>(
```

Comment:
This looks a bit strange, no? But must be reflection love 🤡

Reply:
It's an extra safeguard that only the open generic type is here added.